gluten-flink/ut/src/test/java/org/apache/gluten/table/runtime/stream/custom/NexmarkTest.java

```diff
@@ -0,0 +1,216 @@
+import org.apache.flink.table.api.EnvironmentSettings;
+import org.apache.flink.table.api.TableResult;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.CloseableIterator;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class NexmarkTest {
+
+  private static final Logger LOG = LoggerFactory.getLogger(NexmarkTest.class);
+
+  private StreamExecutionEnvironment env;
+  private StreamTableEnvironment tEnv;
+
+  @BeforeAll
+  public static void setupAll() {
+    LOG.info("NexmarkTest setup");
+    Velox4jEnvironment.initializeOnce();
+  }
+
+  @BeforeEach
+  public void setup() {
+    env = StreamExecutionEnvironment.getExecutionEnvironment();
+    env.setParallelism(1);
+
+    EnvironmentSettings settings = EnvironmentSettings.newInstance().inStreamingMode().build();
+    tEnv = StreamTableEnvironment.create(env, settings);
+  }
+
+  @Test
+  void testNexmarkDDLGeneration() {
+    String createDatagenTable =
+        "CREATE TABLE datagen ("
+            + "  event_type INT,"
+            + "  person ROW<id BIGINT, name STRING, dateTime TIMESTAMP(3)>,"
+            + "  auction ROW<id BIGINT, itemName STRING, dateTime TIMESTAMP(3)>,"
+            + "  bid ROW<auction BIGINT, bidder BIGINT, price BIGINT, dateTime TIMESTAMP(3), extra STRING>"
+            + ") WITH ("
+            + "  'connector' = 'datagen',"
+            + "  'number-of-rows' = '10'"
+            + ")";
+
+    TableResult result = tEnv.executeSql(createDatagenTable);
+    assertThat(result.getJobClient().isPresent()).isFalse();
+
+    String createBidTable =
+        "CREATE TABLE bid ("
+            + "  auction BIGINT,"
+            + "  bidder BIGINT,"
+            + "  price BIGINT,"
+            + "  dateTime TIMESTAMP(3),"
+            + "  extra STRING,"
+            + "  WATERMARK FOR dateTime AS dateTime - INTERVAL '4' SECOND"
+            + ") WITH ("
+            + "  'connector' = 'datagen',"
+            + "  'number-of-rows' = '5'"
+            + ")";
+
+    TableResult bidResult = tEnv.executeSql(createBidTable);
+    assertThat(bidResult.getJobClient().isPresent()).isFalse();
+
+    String describeQuery = "DESCRIBE datagen";
+    TableResult describeResult = tEnv.executeSql(describeQuery);
+
+    List<Row> schemaRows = new ArrayList<>();
+    try (CloseableIterator<Row> iterator = describeResult.collect()) {
+      while (iterator.hasNext()) {
+        schemaRows.add(iterator.next());
+      }
+    } catch (Exception e) {
+      LOG.error("Error collecting schema info", e);
+    }
+
+    assertThat(schemaRows).hasSize(4);
+  }
+
+  @Test
+  void testNexmarkQ0Query() {
+    String createSourceTable =
```

Comment:
As above, we can copy the nexmark quiries in files and run them. Check  the supported ones can run without fallback and then read the results to  verify correctness.